src/components/CourseFilter/CourseFilter.tsx

```diff
@@ -0,0 +1,192 @@
+    let updatedTags = [];
+    if (selectedTags.includes(tagName)) {
+      updatedTags = selectedTags.filter((tag) => tag !== tagName);
+      setSelectedTags(updatedTags);
+    } else {
+      updatedTags = [...selectedTags, tagName];
+      setSelectedTags(updatedTags);
+    }
+    if (tagName === '') {
+      setSelectedTags([]);
+    }
+    const tagsAsaString = updatedTags.filter((tag) => tag).join(',');
+    router.push(pathname + '?' + createQueryString('courseTag', tagsAsaString));
+  };
+
+  const handleDateChange = async (range: [any, any]) => {
+    const [startDate, endDate] = range;
+    setStartDate(startDate);
+    setEndDate(endDate);
+    if (startDate === null && endDate === null) {
+      router.push(pathname + '?' + createQueryString('courseDates', ''));
+    } else {
+      router.push(
+        pathname +
+          '?' +
+          createQueryString('courseDates', startDate + '-' + endDate)
+      );
+    }
+  };
+
+  const handleClearSearch = async () => {
+    await handleNameChange(null);
+    await handleTagChange('');
+    await handleDateChange([null, null]);
+    router.replace(pathname);
+  };
+
+  const customStyles = {
+    width: '250px',
+    height: '55px',
+    fontSize: '16px',
+  };
+
+  return (
+    <>
+      <Typography variant="h4">Search courses</Typography>
+      <Box
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          padding: '16px',
+          backgroundColor: palette.white.main,
+        }}
+      >
+        <Button
+          variant="contained"
+          onClick={() => handleClearSearch()}
+          sx={{
+            width: '200px',
+            marginRight: '50px',
+            backgroundColor: palette.surface.main,
+          }}
+        >
+          Clear search
+        </Button>
+        <div>
+          <Autocomplete
+            value={courseName || null}
+            clearOnEscape
+            disablePortal
+            data-testid="search-autocomplete"
+            id="combo-box"
+            options={initialCourses.map((course) => course.name)}
+            renderOption={(props, option) => {
+              return (
+                <li {...props} key={option}>
+                  {option}
+                </li>
+              );
+            }}
+            sx={{ width: '250px', marginRight: '50px' }}
+            renderInput={(value) => (
+              <TextField {...value} label="Search by a name" />
+            )}
+            onChange={(event, value) => {
+              handleNameChange(value);
+            }}
+          />
+        </div>
+        <DatePicker
+          fixedHeight
+          placeholderText="  Search by dates"
+          minDate={new Date()}
+          selected={startDate}
+          onChange={handleDateChange}
+          startDate={startDate}
+          endDate={endDate}
+          selectsRange
+          showWeekNumbers
+          isClearable
+          highlightDates={initialCourses.map((course) => course.startDate)}
```

Comment:
I was wondering about this part.

The highlighted dates might be unclear to the user since there is no information about them. I was looking at the calendar for a while and didn't figure out immediately that these are the start dates. Should we keep this? You could ask someone else to check if the highlighted dates are self-explanatory to the user... if not, might consider removing.